### DISCUSSION TITLE
zoom: links open in firejailed firefox but downloads are not saved

### DISCUSSION THREAD

**Discussion by bouncinglime** [netblue30/firejail#4570] [Q&A] [answered]
Sorry for the long title... I wasn't sure how else to describe it. Feel free to suggest something more better and I'll update it.

Steps to reproduce :
1. Have firejail and zoom installed on Pop_OS and up to date.
2. Run `firejail zoom`.
3. When someone sends a link in the zoom chat, click the link. [See Result 1.]
4. Run `firejail --list` to ensure that there's now a firefox instance listed.
5. Download a file in that firefox instance to your Downloads directory. [See Result 2.]
6. Navigate to your downloads folder. [See Result 3.]

Results :
1. A new instance of firefox opens - I can tell because it's a fresh profile with the standard start page.
2. Only the Downloads and `/opt/zoom` directories are visible.
3. Files are nowhere to be found. Where did they go?

**Comment by rusty-snake** [comment, marked as answer]
> Where did they go?

`tmpfs` in your RAM and destroyed when the sandbox is closed.

> Only the Downloads and /opt/zoom directories are visible.

I don't think you Downloads directory is visible. If you `touch ~/Downloads/test01` outside of an sandbox and then look at `~/Downloads` inside the sandbox, it should not appear.

**Reply by bouncinglime** [reply]
Thanks for the info. I was also not able to find the files even while the sandbox was still running, though. I was trying to save the linked file - is there any way to do that from within the sandbox environment?

**Reply by rusty-snake** [reply]
1. `whitelist ${DOWNLOADS}`
2. Use file-transfer
3. Copy&Paste the link to curl/wget/firefox if it does not require cookies.
